keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+To enable this feature requires adding to the Kubelet yaml file for the CPUManager feature gate and add the policy option for L3 Cache alignment
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+Cannot be dynamically enabled/disabled because of the CPU Manager state
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyOptions`
+  - Components depending on the feature gate: `kubelet`
+- [x] Other
+  - Describe the mechanism: Change the `kubelet` configuration to set a `CPUManager` policy of static then setting the policy option of `align-cpus-by-uncorecache`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes, a `kubelet` restart is required for changes to take place.
+
+###### Does enabling the feature change any default behavior?
+
+No, to enable this feature, it must be explicitly set in the `CPUManager` static policy and the policy option `align-cpus-by-uncorecache` must be set.
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled it will just require a restart of `kubelet`. The Kubelet configuration will need to be set with the static policy option and align-cpus-by-uncorecache flag removed.
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Feature will be enabled. Proper drain of node and restart of kubelet required. Feature is not intended to be enabled/disabled dynamically, similar to static policy.
+
+###### Are there any tests for feature enablement/disablement?
+
+No, not intended to be enabled/disabled dynamically. 
```

Comment:
This should be yes. You should have unit tests that cover if the policy is added and not.

In this case, think about what kind of testing you will need if the policy does not exist and if the policy exists.

Reply:
I'm not following the question is we have the feature enabled/disabled and we are saying that this should can't really be done because it requires a restart of kubelet and the deletion of the CPU Manager state file.

Reply:
However, unit tests in each component dealing with managing data, created
with and without the feature, are necessary.

So I was thinking you’d have a feature gate which I’m not sure if you are considering one..

Either way you should have tests based on this policy and tests that cover if this policy isn’t detected.

Reply:
another option is the e2e_node serial suite, which would allow for a kubelet to be restarted. I wouldn't consider it a requirement but you could potentially use that

Reply:
understood, added the feedback to the kep